src/java/org/apache/cassandra/index/sai/plan/StorageAttachedIndexQueryPlan.java

```diff
@@ -122,6 +126,23 @@ public Index.Searcher searcherFor(ReadCommand command)
                                                 DatabaseDescriptor.getRangeRpcTimeout(TimeUnit.MILLISECONDS));
     }
 
+    /**
+     * Called on coordinator after merging replica responses before returning to client
+     */
+    @Override
+    public Function<PartitionIterator, PartitionIterator> postProcessor(ReadCommand command)
+    {
+        if (!isTopK())
+            return partitions -> partitions;
+
+        // if there is no limit, just return all rows
+        if (command.limits().isUnlimited())
```

Comment:
If this is a top-k query can it be unlimited?

Reply:
if user doesn't provide limit and page size, it will be unlimited. Perhaps it should be capped as `MAX_TOP_K` in case it's unlimited?